File: src/main/java/com/jcloisterzone/action/ReturnMeepleAction.java
package com.jcloisterzone.action;

import com.jcloisterzone.board.pointer.MeeplePointer;
import com.jcloisterzone.game.ReturnMeepleSource;
import io.vavr.collection.Set;

public class ReturnMeepleAction extends AbstractPlayerAction<MeeplePointer> {

    private final ReturnMeepleSource returnMeepleSource;

    public ReturnMeepleAction(Set<MeeplePointer> options, ReturnMeepleSource returnMeepleSource) {
        super(options);
        this.returnMeepleSource = returnMeepleSource;
    }

    public ReturnMeepleSource getReturnMeepleSource() {
        return returnMeepleSource;
    }
}

File: src/main/ts/com/jcloisterzone/action/ReturnMeepleAction.ts
import type { Set } from "../../../io/vavr/Set.js";
import type { MeeplePointer } from "../board/pointer/MeeplePointer.js";
import type { ReturnMeepleSource } from "../game/ReturnMeepleSource.js";
import { AbstractPlayerAction } from "./AbstractPlayerAction.js";

/** Offer to return a deployed meeple (e.g. voluntary abbot return to score). */
export class ReturnMeepleAction extends AbstractPlayerAction<MeeplePointer> {
  static readonly simpleName = "ReturnMeepleAction";

  constructor(
    options: Set<MeeplePointer>,
    private readonly returnMeepleSource: ReturnMeepleSource,
  ) {
    super(options);
  }

  getReturnMeepleSource(): ReturnMeepleSource {
    return this.returnMeepleSource;
  }
}
